sdcm/sct_provision/instances_provider.py

```
@@ -68,7 +80,20 @@ def provision_instances_with_fallback(
         }
         remoter = RemoteCmdRunnerBase.create_remoter(**ssh_login_info)
         wait_cloud_init_completes(remoter=remoter, instance=v_m)
-        # todo: wait for scylla-machine-image service to complete if instance is scylla-db?
+        # TODO: wait for scylla-machine-image service to complete if instance is scylla-db?
+
+    # Instances are created in parallel by the provisioner, but each node still needs an SSH
+    # connection and a (potentially multi-minute) cloud-init wait. Doing that serially made
+    # provisioning time grow linearly with cluster size, so wait for all nodes concurrently.
+    instance_pairs = list(zip(definitions, provisioned_instances))
```

Ruff (0.16.0)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)